### PR TITLE
Fix #268: added platform support to Catena 4802

### LINKED ISSUE #268: Add support for Catena 4802
Add platform support for new device Catena 4802.

### PR DESCRIPTION
**Pull request by dhineshkumarmcci**
Added support to new board Catena 4802, changes done:
- added ReadVin() to read analog value
- added GUID for 4802 base
- added I2C expander to platform table
- updated version number to v0.19.10.0
- updated README with 4802 and version details
- added 4802 to class hierarchy diagram

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -136,7 +136,7 @@ script:
  - _notsamd    || arduino --verify $(_samdopts '' in866) $THISLIB/examples/catena_hello_lora/catena_hello_lora.ino
 
  #
- # *** TESTS FOR STM32L0 / Catena 4551. 46xx boards ****
+ # *** TESTS FOR STM32L0 / Catena 4551. 46xx. 480x boards ****
```
Comment (terrillmoore) [inline .travis.yml:139]: This is duplicating a typo -- the '.' should have been a ',' in the first place. Let's fix this. If everything I find is cosmetic, I'll do it.

File: README.md [hunk 2]
```diff
@@ -1422,6 +1433,7 @@ This sketch demonstrates the use of the Catena FSM class to implement the `Turns
 
 - HEAD includes the following changes.
 
+  - Fix [#268](https://github.com/mcci-catena/Catena-Arduino-Platform/issues/268): adds support to the new board Catena 4802 (version 0.19.10.0).
```
Comment (terrillmoore) [inline README.md:1436]: This is not a patch, as it's adding new features. Either we have to make this 0.19.0.40 (a non-Semantic-Version field) or this has to be v0.20.0.  It's easier if this is a pre-release, and we do a separate release ticket. So need to be 0.19.0.40.

Generally, you can never change *any* of the three standard semantic-version fields on a branch or PR.

File: src/Catena4802.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	// uses default destructor
+
+	// neither copyable nor movable
+	Catena4802(const Catena4802&) = delete;
+	Catena4802& operator=(const Catena4802&) = delete;
+	Catena4802(const Catena4802&&) = delete;
+	Catena4802& operator=(const Catena4802&&) = delete;
+
+	virtual const char *CatenaName() const override { return "Catena 4802"; };
+#if defined(ARDUINO_MCCI_CATENA_4802)
```
Comment (terrillmoore) [inline src/Catena4802.h:44]: This is a problem. It's already conditioned above, so this should not be needed. Please justify use of a conditional compile here.

File: src/Catena480x.h [hunk 4]
```diff
@@ -70,6 +70,9 @@ class Catena480x : public CatenaStm32L0
 	enum ANALOG_PINS
 		{
 		APIN_VBAT_SENSE = A0,
+#if defined(ARDUINO_MCCI_CATENA_4802)
+		APIN_VIN_SENSE = A0,
```
Comment (terrillmoore) [inline src/Catena480x.h:74]: Forbidden to have this kind of dependency in a higher level document. Must refactor.

File: src/Catena480x.h [hunk 5]
```diff
@@ -80,6 +83,9 @@ class Catena480x : public CatenaStm32L0
 		ANALOG_CHANNEL_A3 = 3,
 		ANALOG_CHANNEL_A4 = 2,
 		ANALOG_CHANNEL_VBAT = ANALOG_CHANNEL_A0,
+#if defined(ARDUINO_MCCI_CATENA_4802)
```
Comment (terrillmoore) [inline src/Catena480x.h:86]: Forbidden construct.

File: src/CatenaBase.h [hunk 6]
```diff
@@ -55,7 +55,7 @@ Copyright notice:
 #define CATENA_ARDUINO_PLATFORM_VERSION_CALC(major, minor, patch, local)        \
         (((major) << 24u) | ((minor) << 16u) | ((patch) << 8u) | (local))
 
-#define CATENA_ARDUINO_PLATFORM_VERSION CATENA_ARDUINO_PLATFORM_VERSION_CALC(0, 19, 0, 30)      /* v0.19.0.30 */
+#define CATENA_ARDUINO_PLATFORM_VERSION CATENA_ARDUINO_PLATFORM_VERSION_CALC(0, 19, 10, 0)      /* v0.19.10.0 */
```
Comment (terrillmoore) [inline src/CatenaBase.h:58]: See previous comment, this is not allowed for a branch.

File: src/CatenaStm32L0.h [hunk 7]
```diff
@@ -71,6 +71,9 @@ class CatenaStm32L0 : public CatenaStm32
         // read the current battery voltage, in engineering units
         virtual float ReadVbat(void) const = 0;
         virtual float ReadVbus(void) const = 0;
+#if defined(ARDUINO_MCCI_CATENA_4802)
```
Comment (terrillmoore) [inline src/CatenaStm32L0.h:74]: Not permitted.

File: src/lib/stm32/catena480x/Catena4802_ReadVoltage.cpp [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+
+/****************************************************************************\
+|
+|	Variables.
+|
+\****************************************************************************/
+
+float
+Catena4802::ReadVin(void) const
```
Comment (terrillmoore) [inline src/lib/stm32/catena480x/Catena4802_ReadVoltage.cpp:49]: This is probably not going to work given that there are header file issues and refactoring will be needed.

**Review by terrillmoore — CHANGES_REQUESTED** [review]
Sorry ... the `#ifs` require that this be reworked. It's architecturally not permitted to add board-specific dependencies in higher-level include files. So this can't be merged as is.

The best policy is not ever to use `#if` unless there is no alternative due to the IDE limitations. That's not the case here; a combination of virtual methods and careful hierarchy will suffice.

File: src/Catena4802.h (hunk 3 above)
Comment (dhineshkumarmcci) [reply src/Catena4802.h:44]: This is added by mistake, and will be removing it.

**Comment by dhineshkumarmcci** [comment]
made the suggested changes and tested build with example sketches. They compile successfully

**Review by terrillmoore — APPROVED** [review]
Looks good.

**Pull request merged**